robot/impl/resource_manager_test.go

```diff
@@ -1850,3 +1852,76 @@ func managerForDummyRobot(t *testing.T, robot robot.Robot) *resourceManager {
 	}
 	return manager
 }
+
+// TestReconfigureParity validates that calling the synchronous and asynchronous version
+// of `Reconfigure` results in the same resource state on a local robot for all
+// combinations of an initial and updated configuration.
+func TestReconfigureParity(t *testing.T) {
+	files := []string{
+		"data/diff_config_deps1.json",
+		"data/diff_config_deps2.json",
+		"data/diff_config_deps3.json",
+		"data/diff_config_deps4.json",
+		"data/diff_config_deps5.json",
+		"data/diff_config_deps6.json",
+		"data/diff_config_deps7.json",
+		"data/diff_config_deps8.json",
+		"data/diff_config_deps9_good.json",
+		"data/diff_config_deps9_bad.json",
+		"data/diff_config_deps10.json",
+		"data/diff_config_deps11.json",
+		"data/diff_config_deps12.json",
+	}
+	type fileConf struct {
+		file string
+		conf *config.Config
+	}
+	var fcs []fileConf
+	for _, file := range files {
+		fcs = append(fcs, fileConf{file: file, conf: ConfigFromFile(t, file)})
+	}
+
+	logger := logging.NewTestLogger(t)
+	ctx := context.Background()
+
+	testReconfigureParity := func(t *testing.T, from fileConf, to fileConf) {
+		name := fmt.Sprintf("%s -> %s", from.file, to.file)
+		t.Run(name, func(t *testing.T) {
+			// TODO: resolve data races and to allow this to run in parallel
+			// t.Parallel()
```

Comment:
assuming we want to keep this test, we definitely want to figure this out since the test takes over a minute to run otherwise